### PR TITLE
Feat: ArgoCD addon

### PR DESCRIPTION
**Pull request by cezhang**
Signed-off-by: cezhang <c1zhang.dev@gmail.com>

### Checklist

I have:
- [ ] Title of the PR starts with OAM type (e.g. `[Workload]` or `[Trait]` or `[Scope]`).
- [ ] Updated/Added any relevant [documentation] and [examples].
- [ ] Unit/E2E Tests added.

### CODE REVIEW

**Review by wangyikewxgm — COMMENTED** [review]
Absolutely cool job！Thanks for your contribution

File: experimental/addons/argocd/definitions/application.yaml [hunk 1]
```diff
@@ -0,0 +1,126 @@
+# Code generated by KubeVela templates. DO NOT EDIT. Please edit the original cue file.
+apiVersion: core.oam.dev/v1beta1
+kind: ComponentDefinition
+metadata:
+  annotations:
+    definition.oam.dev/description: Argocd-application is a definition of application resource.
```
Comment (wangyikewxgm) [inline experimental/addons/argocd/definitions/application.yaml:6]: There is no need put both cue and yaml in this dir. The cue file can be translate to x-defn k8s-object
automatically and apply to cluster.

Comment (cezhang) [reply experimental/addons/argocd/definitions/application.yaml:6]: I think this is first version, coming up modification may be needed, so i just keep it here first. Should I delete those and commit again?

Comment (wangyikewxgm) [reply experimental/addons/argocd/definitions/application.yaml:6]: Yes I suggest  keeping the cue files is enough.

File: experimental/addons/argocd/metadata.yaml [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+url: "https://github.com/argoproj/argo-cd"
+
+tags:
+  - argocd
+  - gitops
+  - v2.2.5
+
+deployTo:
+  control_plane: true
+
```
Comment (wangyikewxgm) [inline experimental/addons/argocd/metadata.yaml:14]: I think this addon should be used in runtime cluster. Please add `runtimeCluster: true`. And `control_plane` is useless please remove this line.

File: experimental/addons/argocd/definitions/application.cue [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+
+		operation?: #operation
+	}
+
+	output: {
+		apiVersion: "argoproj.io/v1alpha1"
+		kind:       "Application"
+		metadata: {
+			name:      parameter.name
+			namespace: context.namespace
```
Comment (wangyikewxgm) [inline experimental/addons/argocd/definitions/application.cue:112]: Should we set namespace is "arogcd". Let's "argo-app" always in the same namespace?

Comment (cezhang) [reply experimental/addons/argocd/definitions/application.cue:112]: Yes, the argocd application controller only watches those CRD resources in the same namespace

Comment (wangyikewxgm) [inline experimental/addons/argocd/definitions/application.cue:112]:
```suggestion
			namespace: argocd
```

**Review by wangyikewxgm — COMMENTED** [review]
experimental addon won't be checked by the CI workflow, please add test for this addon like https://github.com/oam-dev/catalog/blob/5a7460ab5b9750f747c6de88ffef547fce1c7fd3/test/e2e-test/hack/addon-vela-test.sh#L22

File: experimental/addons/argocd/definitions/application.cue [hunk 4]
```diff
@@ -0,0 +1,131 @@
+"argocd-application": {
```
Comment (wonderflow) [inline experimental/addons/argocd/definitions/application.cue:1]: Let's name it by `argo-app` it's more clear

**Comment by wangyikewxgm** [comment]
LGTM, Thanks. please fix the last comment.

**Review by wonderflow — APPROVED** [review]

**Review by wangyikewxgm — APPROVED** [review]

**Pull request merged**